include/swift/AST/DiagnosticsClangImporter.def

```diff
@@ -289,6 +289,9 @@ ERROR(both_returns_retained_returns_unretained, none,
       "SWIFT_RETURNS_UNRETAINED",
       (const clang::NamedDecl *))
 
+ERROR(redundant_conformance_protocol,none,
+     "redundant conformance of %0 to protocol '%1'", (Type, StringRef))
```

Comment:
I'm not sure if redundant conformances should trigger an error. There are legitimate cases where you would end up with redundant conformances, for instance:
```cpp
struct Base1 __attribute__((swift_attr("conforms_to:Mod.Proto"))) {};
struct Base2 __attribute__((swift_attr("conforms_to:Mod.Proto"))) {};
struct Derived : Base1, Base2 {};
```
Would the compiler emit an error for this code?

Reply:
Thank you for your review.  I have added a flag to skip the checks for protocol conformance originating from bases.  With this change, redundant conformance errors will only occur when a type itself declares conformance to the same protocol multiple times.

Reply:
> it's more difficult to convert an error into a warning

Could you elaborate on this? I'm not quite following what this means

Reply:
Oops, I meant to say, it's _less_ difficult to convert an error into a warning.

Reply:
Aha! Agreed